shared/js/shrinking_ui.js

```diff
@@ -313,21 +312,27 @@
   ShrinkingUI.prototype._sendingSlideTo =
     function su_sendingSlideTo(y, callback) {
       var foregroundElement = this.elements.foregroundElement;
+      var cbDone = (function on_cbDone(evt) {
+        foregroundElement.removeEventListener('transitionend', cbDone);
+        this.elements.foregroundElement.style.transitionDuration = '0s';
+        if ('undefined' !== typeof callback) {
+          callback();
+        }
+      }).bind(this);
       if ('TOP' === y) {
+        foregroundElement.style.transitionDuration = '0.3s';
         y = foregroundElement.parentElement.clientHeight;
       } else if ('BOTTOM' === y) {
+        if (this.state.touch.prevY > 0) {
```

Comment:
We should add comments to explain why the transitionDuration jumps between 0.0s and 0.3s, and what the `this.state.touch.preY > 0` means.